dashboard/src/hooks/use-is-changed.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+
+export default function useIsChanged<T extends Record<string, any>>(
+  currentData: T | null | undefined,
+  originalData: T | null | undefined,
+): boolean {
+  const [isChanged, setIsChanged] = useState(false);
+
+  useEffect(() => {
+    if (!originalData || !currentData) {
+      setIsChanged(false);
+      return;
+    }
+
+    const keys = Object.keys(currentData) as (keyof T)[];
+    const changes = keys.some((key) => currentData[key] !== originalData[key]);
+    setIsChanged(changes);
+  }, [currentData, originalData]);
+
+  return isChanged;
+}
```

Comment:
Nitpick: This can be simplified using `useMemo` 😄 Not a blocker by any means though!

```suggestion
export default function useIsChanged<T extends Record<string, any>>(
  currentData: T | null | undefined,
  originalData: T | null | undefined,
): boolean {
  return useMemo(() => {
    if (!originalData || !currentData) {
      return false;
    }
    const keys = Object.keys(currentData) as (keyof T)[];
    return keys.some((key) => currentData[key] !== originalData[key]);
  }, [currentData, originalData]);
}
```

Reply:
Done in f6d14c38007b81ca115d311dfd6fc1608849919e 👌🏻